kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/KafkaProxyContext.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import java.time.Clock;
+import java.util.stream.Collectors;
+
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+import io.kroxylicious.kubernetes.api.v1alpha1.VirtualKafkaCluster;
+import io.kroxylicious.kubernetes.operator.model.ProxyModel;
+
+public record KafkaProxyContext(Clock clock,
+                                SecureConfigInterpolator secureConfigInterpolator,
+                                ProxyModel model) {
+
+    private static final String KEY_CTX = KafkaProxyContext.class.getName();
+
+    static void init(Context<KafkaProxy> context,
+                     Clock clock,
+                     SecureConfigInterpolator secureConfigInterpolator,
+                     ProxyModel model) {
+        var rc = context.managedWorkflowAndDependentResourceContext();
+
+        rc.put(KEY_CTX,
+                new KafkaProxyContext(
+                        Clock.fixed(clock.instant(), clock.getZone()),
+                        secureConfigInterpolator,
+                        model));
+    }
+
+    static KafkaProxyContext proxyContext(Context<KafkaProxy> context) {
+        return context.managedWorkflowAndDependentResourceContext().getMandatory(KEY_CTX, KafkaProxyContext.class);
+    }
+
+    boolean isBroken(VirtualKafkaCluster cluster) {
```

Comment:
I guess we were missing a test somewhere? checking that a cluster with unresolved dependencies and ingress exceptions is broken.